Eliminate OpenAI -> Gemini Model Mapping

ekatiyar opened the pull request:
Addresses #35 

This change eliminates the mapping of openAI models to gemini models and directly exposes the underlying gemini models to the api endpoints directly. This was motivated by my own issues with using Google's [openAI API Compatible Endpoint](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/call-gemini-using-openai-library).

My own usecase is flexible enough to use non-openAI named models with the OpenAI API, so the model mapping from openAI model names to gemini model names just serves to confuse things.

I recognize this may not be the case for all users of this project and some clients may only be able to handle OpenAI-named models; in that case feel free to reject this PR.

ekatiyar committed:
Eliminate OpenAI -> Gemini Model Mapping

ekatiyar committed:
Update README

ekatiyar commented:
If this is something you are interested in, I can create a separate branch without the readme changes

ekatiyar committed:
Add Support for Embeddings Endpoint

ekatiyar committed:
Update README with Embeddings Usage

zhu327 commented on README.md:
```diff
@@ -1,6 +1,8 @@
-Gemini-OpenAI-Proxy is a proxy designed to convert the OpenAI API protocol to the Google Gemini Pro protocol. This enables seamless integration of OpenAI-powered functionalities into applications using the Gemini Pro protocol.
+Gemini-OpenAI-Proxy is a proxy designed to convert the OpenAI API protocol to the Google Gemini protocol. This enables applications built for the OpenAI API to seamlessly communicate with the Gemini protocol, including support for Chat Completion, Embeddings, and Model(s) endpoints.
+
+This is a fork of zhu327/gemini-openai-proxy that eliminates the mapping of openAI models to gemini models and directly exposes the underlying gemini models to the api endpoints directly. I've also added support for Google's embeddings model. This was motivated by my own issues with using Google's [openAI API Compatible Endpoint](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/ca ...
```
Remove the steps about forking from the README.

zhu327 commented on README.md:
````diff
@@ -30,11 +32,24 @@ go build -o gemini main.go
-```bash
-docker run --restart=always -it -d -p 8080:8080 --name gemini zhu327/gemini-openai-proxy:latest
-```
+   You can either do this on the command line:
````
GitHub Actions will automatically build Docker images, so we can just use the existing image here.

ekatiyar commented:
This is just the main branch of my fork, which is kinda messy since I've added some other stuff since I opened this pr. I can reorganize it so its easier to merge into your repo, but to start with I'd think I'd want to merge in the embeddings support I added in #39 , and then create a separate branch for eliminating the model mapping

ekatiyar commented:
Closing this PR in favor of merging in #41 which accomplishes the same thing in a cleaner and non-disruptive way
